ci/cached-builds/gen_gha_matrix_jobs.py

```diff
@@ -108,6 +108,10 @@ def write_github_workflow_file(tree: dict[str, list[str]], path: pathlib.Path) -
             "workflow_dispatch": {},
             "schedule": [{ "cron": "0 2 * * *"}], # 2am UTC everyday
         },
+        "concurrency": {
+            "group": "${{ format('build-notebooks-{0}', github.sha) }}",
```

Comment:
When does it happen that there is more than one run at the same time with the same sha? Sounds unlikely to me.

Reply:
I can think of two scenarios: 1) subsequent `workflow_dispatch` and 2) when a `schedule` starts after a running `workflow_dispatch` or vice-versa.

Reply:
On second thought, these scenarios should be rare and I don't want the `schedule` to be canceled if a `workflow_dispatch` starts right after. I'm gonna revert that.